### ISSUE TITLE
"Add installation path to PATH env" function

### ISSUE THREAD

**Issue by soredake** [microsoft/winget-cli#4008]
### Description of the new feature / enhancement

There is some software that install cli version's but do not modify PATH env, it's not realistic to expect that every installer will be updated to modify, therefore having such function in winget will be nice.

Example packages that have cli version's but don't add installation path to PATH env so you can't use them out of the box:
[proxinject](https://github.com/PragmaTwice/proxinject/issues/13)
[7-Zip-zstd](https://github.com/mcmilk/7-Zip-zstd/issues/360)
7zip
this issue have many more example of such software https://github.com/microsoft/winget-cli/issues/549

### Proposed technical implementation details

Example for proxinject package:
```
Installers:
- Architecture: x64
  InstallerUrl: https://github.com/PragmaTwice/proxinject/releases/download/v0.5.0-pre/proxinjectSetup-v0.5.0-pre-x64.exe
  InstallerSha256: 9F8D33D6FEAE2D3042D01F129204318885178A9363EA9AAF468FABE9F4E2F9E2
  AddToPath: $env:APPDATA\proxinject | <INSTALLPATH>
```

AddToPath should only work in Silent or SilentWithProgress modes as user can modify installation path in interactive mode and winget will add wrong path to env.

If user is not specified `--location` argument it will use default value `$env:APPDATA\proxinject` otherwise it will add `<INSTALLPATH>` to `PATH`

**Comment by Trenly** [comment]
What if an installer doesn’t write to `$env:APPDATA` and `<INSTALLPATH>` isn’t set?

**Comment by soredake** [comment]
@Trenly then it will not work, yes.

**Comment by stephengillie** [comment]
I believe this functionality might already exist within the client.

**Comment by lesleyrs** [comment]
Winget did install some cli tools to PATH for me, but `sharkdp.bat` and `sharkdp.fd` are 2 that didn't get added automatically even though they also got installed to `AppData\Local\Microsoft\WinGet\Packages` just like the others. IDK what the problem is though.

**Comment by denelon** [comment]
WinGet isn't going to take action to modify environment variables like the path unless it's a portable package where WinGet is logically acting as an installer. There is too much risk of undesired behavior, and if the package is uninstalled, it could lead to dead path entries.

It might be worth adding installation notes to manifests for those packages so users can be informed what action they need to take.

**Comment by sohang3112** [comment]
> WinGet isn't going to take action to modify environment variables like the path

@denelon if it's not going to add to the environment path, at least it can tell *what* the executable path is so that it can be added to environment path manually!

For example, `pip` also warns while installing packages if install path is not in environment path.

**Comment by HeitorAugustoLN** [comment]
@denelon Adding the feature to add the installation to PATH during installation is very important for some portable applications.  For example, when I created a manifest file for [Dart](https://dart.dev) it had some issues such as the ‘Failed to start the Dart CLI isolate (null)’ error when trying to run it. However, when I manually added the installation path to the system environment PATH, it worked flawlessly.
